src/config-context.js

```diff
@@ -1,26 +1,28 @@
-import React, { useContext, useState } from 'react';
+import React, { useContext, useMemo, useState } from 'react';
 import PropTypes from 'prop-types';
 
 import defaultConfig from './config';
 
 const ConfigContext = React.createContext({});
 const ConfigConsumer = ConfigContext.Consumer;
 
-const ConfigProvider = ({ children, config: customConfig }) => {
-  const [config, setConfig] = useState({ ...defaultConfig, ...customConfig });
+const ConfigProvider = ({ children, config: customConfig, defaultTheme }) => {
+  const [activeTheme, setActiveTheme] = useState(defaultTheme);
 
-  const setTheme = (newTheme = '') => {
-    setConfig({
-      ...config,
-      activeTheme: newTheme
-    });
-  };
+  const theme = useMemo(() => {
+    const config = { ...defaultConfig, ...customConfig };
+    const defaultSpacerSize = config?.[activeTheme]?.defaultSpacerSize || 2;
```

Comment:
Let's maybe set the defaults from `defaultTheme`. What do you think?
```suggestion
   const defaultSpacerSize = config?.[activeTheme]?.defaultSpacerSize || defaultConfig.default.defaultSpacerSize;
```